### PR TITLE
Allow section content to be function

### PR DESCRIPTION
**Pull request by eragon512**
Marking this PR as a WIP, as I am not sure how to refactor to maintain consistency across the project

Refers: #1224 

Edit: Removing WIP status, this PR is good to go

### CODE REVIEW

File: src/loaders/utils/getSections.js [hunk 1]
```diff
@@ -10,6 +10,38 @@ const slugger = require('./slugger');
...
+
+	if (_.isFunction(section.content)) {
+		return {
+			content: [
+				{
+					type: 'markdown',
+					content: section.content(),
+				},
+			],
+			contentRelativePath,
```
Comment (sapegin) [inline src/loaders/utils/getSections.js]: You need to test that but looks like `filepath` prop isn't used anywhere in sections. It's passed here but I don't see where it's used:

https://github.com/styleguidist/react-styleguidist/blob/34f3c83e769542157c72d0e055ad8850d22b6001/src/client/rsg-components/Section/SectionRenderer.js#L52

File: src/loaders/utils/getSections.js [hunk 2]
```diff
@@ -10,6 +10,29 @@ const slugger = require('./slugger');
...
+			content: section.content(),
+		};
+	}
+
+	// Try to load section content file
+	const contentAbsolutePath = path.resolve(config.configDir, contentRelativePath);
+	if (!fs.existsSync(contentAbsolutePath)) {
+		throw new Error(`Styleguidist: Section content file not found: ${contentAbsolutePath}`);
+	}
+	const content = requireIt(`!!${examplesLoader}!${contentAbsolutePath}`);
```
Comment (sapegin) [inline src/loaders/utils/getSections.js]:
```suggestion
	return requireIt(`!!${examplesLoader}!${contentAbsolutePath}`);
```

File: src/loaders/utils/getSections.js [hunk 3]
```diff
@@ -10,6 +10,29 @@ const slugger = require('./slugger');
...
+	return content;
```
Comment (sapegin) [inline src/loaders/utils/getSections.js]:
```suggestion
```

**Comment by eragon512** [comment]
@sapegin this PR is good to go, added your requested changes

**Comment by eragon512** [comment]
hey @sapegin , any more changes needed to this PR?

**Comment by sapegin** [comment]
Thanks!

**Pull request merged**